### PR TITLE
fix checker no notes needed

### PR DESCRIPTION
**Pull request by earonesty**
fix checker no notes

### CODE REVIEW

File: rnotes/runner.py [hunk 1]
```diff
@@ -402,12 +410,25 @@ def branch_check(self):
...
+        all_diff = self.git("diff", "--name-only", diff_base)
+        for ent in all_diff.split("\n"):
+            ent = ent.strip()
+            if not ent or self.not_important(ent):
+                continue
+            if ent.startswith(self.notes_dir):
+                self.lint_file(ent)
+                continue
+            log.debug("need notes: %s", ent)
+            need_notes = True
```
Comment (colatkinson) [inline rnotes/runner.py:423]: Can break early here

File: rnotes/runner.py [hunk 2]
```diff
@@ -402,12 +410,25 @@ def branch_check(self):
...
             diff_base = target
 
-        diff = self.git("diff", "--name-only", "--diff-filter=A", diff_base)
+        need_notes = False
+        all_diff = self.git("diff", "--name-only", diff_base)
+        for ent in all_diff.split("\n"):
+            ent = ent.strip()
+            if not ent or self.not_important(ent):
+                continue
+            if ent.startswith(self.notes_dir):
```
Comment (colatkinson) [inline rnotes/runner.py:419]: Should this skip deleted files? Should be rare in practice, but I could see a use case if some bad notes made it into trunk and need to be renamed/consolidated/rewritten.

**Review by colatkinson — APPROVED** [review]

File: rnotes/runner.py (hunk 2 above)
Comment (earonesty) [reply rnotes/runner.py:419]: yep

Comment (earonesty) [reply rnotes/runner.py:419]: we skip fnf errs for relnotes, should be enough

**Pull request merged**